How to use the NGINX Docker image (#859)
OctopusDeploy/blog: blog/2022-q3/using-nginx-image/index.md

---
title: Using the NGINX Docker image
description: Learn how to create Docker web apps based on the NGINX image.
author: matthew.casperson@octopus.com
visibility: public
published: 2022-09-06-1400
metaImage: blogimage-gettingstartedcontainerisation-2022.png
bannerImage: blogimage-gettingstartedcontainerisation-2022.png
bannerImageAlt: Man sitting on top of container with green circle with a power up icon
isFeatured: false
tags: 

>   - Containers

Reviewer: ```suggestion
  - DevOps
  - Containers
```

  - Docker
---

Docker is a compelling platform to package and run web applications, especially when paired with one of the many Platform as a Service (PaaS) offerings provided by cloud platforms. NGINX has long provided DevOps teams with the ability to host web applications on Linux, and also provides an official Docker image to use as the base for custom web applications. 

> In this post I'll explain how DevOps teams can use the [NGINX Docker image](https://hub.docker.com/_/nginx) to build and run web applications on Docker.

Reviewer: ```suggestion
In this post, I explain how DevOps teams can use the [NGINX Docker image](https://hub.docker.com/_/nginx) to build and run web applications on Docker.
```

COPY health-check.conf /etc/nginx/conf.d/health-check.conf
```

Build and run the image with the commands:

```bash
docker build . -t mynginx
docker run -p 8080:80 -p 9090:90 mynginx
```

Once again open `http://localhost:9090/nginx-health` or `http://localhost:8080/index.html` to view the web pages. Everything continues to work as it did previously, but your custom image is now much smaller.

## Conclusion

NGINX is a powerful web server, and the official NGINX Docker image allows DevOps teams to host custom web applications in Docker. NGINX also supports advanced scenarios thanks to its ability to read configuration files copied into a custom Docker image. 

In this post, you learned how to create a custom Docker image hosting a static web application, added advanced NGINX configuration files to provide a health check endpoint, and compared the sizes of Debian and Alpine NGINX images.

## Resources

* [Octopus trial](https://octopus.com/start)
* [NGINX Docker Image](https://hub.docker.com/_/nginx)
* [NGINX Docker Image Source Code](https://github.com/nginxinc/docker-nginx)
* [Dockerfile reference](https://docs.docker.com/engine/reference/builder/)

## Learn more

> If you'd like to build and deploy containerized applications to AWS platforms such as EKS and ECS, try the [Octopus Workflow Builder](https://octopusworkflowbuilder.octopus.com/#/). The Builder populates a GitHub repository with a sample application built with GitHub Actions workflows and configures a hosted Octopus instance with sample deployment projects demonstrating best practices such as vulnerability scanning and Infrastructure as Code (IaC). 

Reviewer: ```suggestion
If you want to build and deploy containerized applications to AWS platforms such as EKS and ECS, try the [Octopus Workflow Builder](https://octopusworkflowbuilder.octopus.com/#/). The Builder populates a GitHub repository with a sample application built with GitHub Actions workflows and configures a hosted Octopus instance with sample deployment projects demonstrating best practices such as vulnerability scanning and Infrastructure as Code (IaC). 
```
